mycity/mycity/utilities/crime_incidents_api_utils.py

```diff
@@ -26,38 +30,45 @@ def get_crime_incident_response(address):
     url_parameters = {"sql": _build_query_string(address)}
     logger.debug("Finding crime incidents information for {} using query {}"
         .format(address, url_parameters))
+
     with requests.Session() as session:
-        response = session.get(CRIME_INCIDENTS_SQL_URL, params=url_parameters)
+        if test_session is None:
+            response = session.get(CRIME_INCIDENTS_SQL_URL, params=url_parameters)
+        else:
+            response = test_session.response
```

Comment:
I understand what you are doing with injection for unit testing, but it seems to add some complexity to both the function signature and the code itself by adding branches like this. Is there a cleaner way?

Reply:
Yeah, I agree that it looks ugly.

I think you can pass functions as an argument in Python. Let me try again